### DISCUSSION TITLE
webots_ros2_driver

### DISCUSSION THREAD

**Discussion by ntukenmez3** [cyberbotics/webots#6371] [General]
Hi all, we would like to implement our controller developed by our team on CF2 simulator. We got the following error, could you help us to get rid of that problem, thanks.

**~/webots_ros2/webots_ros2_mavic/launch$** ros2 launch ~/webots_ros2/webots_ros2_mavic/launch/robot_launch.py

[INFO] [launch]: All log files can be found below /home/ntukenmez3/.ros/log/2023-09-05-19-35-21-978685-ae-icps-407120-168970
[INFO] [launch]: Default logging verbosity is set to INFO
[ERROR] [launch]: Caught exception in launch (see debug for traceback): Caught exception when trying to load file of format [py]: No module named 'webots_ros2_driver'

**Comment by omichel** [comment]
Did you explicitly installed the `webots_ros2_driver` module as explained in the documentation?

**Comment by ntukenmez3** [comment]
Thanks for reply, what does ‘explicitly’ mean, it was installed on the computer, how can I show that it was installed on the computer, if you give me some recommendations, I can share it here, thanks.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by omichel** [reply]
I means did you typed these commands?
```bash
sudo apt update
sudo apt install ros-iron-webots-ros2-driver
source /opt/ros/iron/setup.bash
```
As instructed [here](https://docs.ros.org/en/iron/Tutorials/Advanced/Simulators/Webots/Installation-Ubuntu.html).

**Reply by ntukenmez3** [reply]
Yes, we did.

**Reply by ntukenmez3** [reply]
The following example as in ROS2-Webots tutorial. However, it is still same problem.

`**ntukenmez3@ae-icps-407120:~/ros2_ws$** ros2 launch my_package robot_launch.py `

`[INFO] [launch]: All log files can be found below /home/ntukenmez3/.ros/log/2023-09-07-10-29-05-643135-ae-icps-407120-1381651
[INFO] [launch]: Default logging verbosity is set to INFO
[ERROR] [launch]: Caught exception in launch (see debug for traceback): Caught exception when trying to load file of format [py]: No module named 'webots_ros2_driver`

**Reply by omichel** [reply]
Which version of ROS 2 are you using?

**Reply by ntukenmez3** [reply]
humble

**Reply by ntukenmez3** [reply]
By the way, I fixed it, it works well. Thank you for your helpful explanations.
